contracts/PegSwap.sol

```diff
@@ -88,24 +89,39 @@ contract PegSwap is Owned, ReentrancyGuard {
   /**
    * @notice exchanges the source token for target token
    * @param amount count of tokens being swapped
-   * @param source the token that is being given
-   * @param target the token that is being taken
+   * @param source_ the token that is being given
+   * @param target_ the token that is being taken
    */
   function swap(
     uint256 amount,
-    address source,
-    address target
+    address source_,
+    address target_
   )
     external
     nonReentrant()
   {
-    _removeLiquidity(amount, source, target);
-    _addLiquidity(amount, target, source);
+    ERC20 source = ERC20(source_);
+    ERC20 target = ERC20(target_);
 
-    emit TokensSwapped(amount, source, target, msg.sender);
+    uint8 sDecimals = source.decimals();
+    uint8 tDecimals = target.decimals();
 
-    require(ERC20(source).transferFrom(msg.sender, address(this), amount), "transferFrom failed");
-    require(ERC20(target).transfer(msg.sender, amount), "transfer failed");
+    uint256 amountOut;
+    if (sDecimals > tDecimals) {
+      amountOut = amount / (10 ** uint256(sDecimals - tDecimals));
```

Comment:
to keep the naming similar to source and target, let;'s rename this to
`sourceAmount, `targetAmount`. While the func input parameter should be `sourceAmount`